### ISSUE TITLE
Charming Printed Books on Front-End, UX, A11y — Smashing Magazine

### ISSUE THREAD

**Issue by tg-z** [tg-z/web-clip#740]
![](https://cloud.netlifyusercontent.com/assets/344dbf88-fdf9-42bb-adb4-46f01eedd629/a6df8622-1d58-40ea-89c4-c5a89187d8bf/cover-big-opt.png)

At its heart, “Inclusive Components” is a detailed, practical handbook for building fully accessible interfaces. The book examines common interface patterns — accordions, tables, tabs, toggles and everything in-between — through the lens of inclusion. The result is a dozen of fully **accessible and robust patterns** we author, plug in, and use daily. [Jump to table of contents.](https://www.smashingmagazine.com/printed-books/inclusive-components/#bookTOC)

The book features 12 common UI components, broken down in detail, one by one. The in-depth explorations are meticulously illustrated and code examples culminate as **bulletproof code snippets**, applicable to your work right away. Plus a strategy for building accessible interfaces for your own components — all in one book. [Download a free PDF excerpt](http://provide.smashingmagazine.com/eBooks/inclusive-components-free-chapter.pdf) (1.1 MB).

You’ll learn how to build:

*   accessible buttons and toggle buttons,
*   navigation menus and dropdowns,
*   keyboard-friendly tooltips,
*   “dark mode” themes,
*   accessible content sliders,
*   inclusive notifications,
*   semantic sortable data tables,
*   accessible dialogs and modals.

> “Inclusive Components is chock-full of practical and comprehensive advice on building accessible UI. It’s my go-to resource after the official WCAG and ARIA documentation. I’ve found it extremely helpful when building our design system!”

Sarah Federman

Senior front-end developer

### About The Book

Because we often build and deploy under tough deadlines, we tend to break accessibility without even noticing it. Our products become slower, clunkier and more painful to use — often simply unbearable for **keyboard- and screen reader users**, and as such fragile and vulnerable for legal disputes. Let’s fix it.

We’ve teamed up with [Heydon Pickering](http://www.heydonworks.com/), a seasoned front-end developer with a focus on accessibility, to examine common web UI patterns through the lens of inclusion. The result is a dozen of accessible and robust solutions for the patterns we author, plug in, and use every day.

With _Inclusive Components_, we’ve tried to create a very focused **handbook with applicable, long-living solutions** and strategies to create accessible and inclusive interfaces. [Download a free PDF sample](http://provide.smashingmagazine.com/eBooks/inclusive-components-free-chapter.pdf) (PDF, 1.1 MB)

![](https://cloud.netlifyusercontent.com/assets/344dbf88-fdf9-42bb-adb4-46f01eedd629/a6df8622-1d58-40ea-89c4-c5a89187d8bf/cover-big-opt.png)

### Table of Contents

Each chapter tackles a single component, addressing how we might read and interact with it, and how we can make the component more reliable and robust for every single customer.

+

What does it take to make toggle buttons inclusive? To start off, Heydon takes a look at common pitfalls and what you need to keep in mind to do better.

+

You’ll learn how to build an integrated todo list component from the ground up. This doesn’t have to apply only to todo lists but also to making the basic creation and deletion of content inclusive.

+

Menus, dropdowns, subnavigation. There’s a lot of confusion happening around these terms. Why is this happening, why are WAI-ARIA semantics often misused here, how do we properly use “hamburgers” and “navicons,” and what do you need to consider to make your menus keyboard- and screen-reader-accessible?

+

Inclusive design is often about providing the user with the right tool for the job and the right kind of tooltip to go with that tool. In this chapter, we’ll be looking at situations which might call for a tooltip and learn how to formulate inclusive implementations for them.

+

Offering alternative themes often represents a maintenance issue. This chapter explores how to make an efficient and portable React component that allows users to switch a default light theme into “dark mode”.

+

What makes a tabbed interface a tabbed interface is in the ergonomics of its keyboard behavior. Heydon takes you step-by-step through the process of applying ARIA semantics to master the challenges that tabbed interfaces might bring along.

+

Although implementing collapsible sections is rather simple, the interaction does not have a consistent native implementation across browsers. The tips in this chapter will help you turn your collapsible regions into web components that are easy to include as part of larger patterns and in content files.

+

Carousels don’t have to be bad, but we have a culture of making them bad. This chapter explores how you can create something that fulfills a basic purpose without overloading it with features.

+

When web pages undergo changes as you operate them, it’s important that users are kept abreast of changing states. This chapter looks at notification components and how they can increase confidence in the use of web applications, in an inclusive way.

+

Let’s explore what it takes to make your data tables screen reader accessible, responsive, and as ergonomic as possible for everyone. Plus a trick for fixing old layout tables that cause confusion for visually-impaired users.

+

Modal dialogs are often contentious and problematic, so let’s reconsider if they really make sense for your use case and, if yes, examine them in the context of inclusive design thinking and performance.

+

The card component is the last component tackled in the book as it requires the most invention. So in this final chapter, we’ll look into a few permutations of a simple card component and find a balance between sound HTML structure and ergonomic interaction.

![](https://cloud.netlifyusercontent.com/assets/344dbf88-fdf9-42bb-adb4-46f01eedd629/95c0c743-56df-4630-af66-25df0037ec51/heydon-pickering-author-image-opt.png)

### About The Author

Heydon Pickering ([@heydonworks](https://twitter.com/heydonworks)) has worked with The Paciello Group, The BBC, Smashing Magazine, and Bulb Energy as a designer, engineer, writer, editor, and illustrator. He was shortlisted for Designer Of The Year in The Net Awards. Heydon previously wrote [_Inclusive Design Patterns_](https://www.smashingmagazine.com/printed-books/inclusive-front-end-design-patterns/) which sold over 10,000 copies. Proceeds from this title were donated to the ACLU and The Democratic Socialists Of America, to help these organizations fight fascism and create a more inclusive society.

![](https://cloud.netlifyusercontent.com/assets/344dbf88-fdf9-42bb-adb4-46f01eedd629/c60ed56e-7041-4ada-bf8c-d532a9c00fc5/inclusive-components-secondary-image-opt.png)

### Technical Details

*   332 pages
*   Quality hardcover with stitched binding and ribbon page marker
*   ISBN: 978-3-945749-82-1
*   Free worldwide airmail shipping from Germany
*   Released in December 2019

### Frequently Asked Questions

If you have any questions, we are right here to answer them. We love our customers, and we’d love to help you in any way or just listen to your story. So please feel free to [ask questions via @smashingmag on Twitter](https://www.twitter.com/smashingmag) — we’ll get back to you right away. Just in case: here are answers to some frequently asked questions.

How much does shipping cost to my country?

+

No shipping costs — wherever you are in the world! We ship everywhere, worldwide, via **airmail shipping**. What you see is what you pay. No ifs or buts.

Delivery times to my country?

+

All books will be shipped via airmail to keep delivery times as short as possible. You can find the anticipated delivery time for your country in the overview page on [delivery times](https://www.smashingmagazine.com/delivery-times/).

+

Yes, of course! The **eBook is included** with printed books, so you can start reading the eBook right away. All eBooks are available in PDF, ePub and Amazon Kindle formats.

What payment methods are accepted?

+

We accept **PayPal, VISA, MasterCard and American Express**. Of course, we use a secure connection, with 256-bit AES encryption and a green GeoTrust Extended Validation SSL CA certificate. And no, we don’t store your credit card data on our servers. (_Obviously._)

Is there a money-back guarantee?

+

Yes, absolutely! No risk at all — our **100-day full money-back guarantee** keeps you safe. Don’t hesitate to return your purchase. You’ll get your money back without ifs, ands, or buts!

I have a question that is not covered here.

+

Please get in touch with us via the [contact form](https://www.smashingmagazine.com/contact/) or via [@SmashingSupport on Twitter](https://www.twitter.com/SmashingSupport). We would love to help you as soon as we humanly can. 
 https://www.smashingmagazine.com/printed-books/inclusive-components/
